lib/cylc/suite_srv_files_mgr.py

```diff
@@ -318,13 +318,22 @@ def get_suite_rc(self, reg, suite_owner=None):
             self.FILE_BASE_SUITE_RC)
 
     def get_suite_source_dir(self, reg, suite_owner=None):
-        """Return the source directory path of a suite."""
+        """Return the source directory path of a suite.
+
+        Will register un-registered suites located in the cylc run dir.
+        """
         srv_d = self.get_suite_srv_dir(reg, suite_owner)
         fname = os.path.join(srv_d, self.FILE_BASE_SOURCE)
         try:
             source = os.readlink(fname)
         except OSError:
-            raise SuiteServiceFileError("ERROR: Suite not found %s" % reg)
+            suite_d = os.path.dirname(srv_d)
+            if os.path.exists(suite_d):
```

Comment:
Should also check `not is_remote_user(suite_owner)` - only register for current user.